pkg/election/election.go

```diff
@@ -349,10 +388,50 @@ func (e *Election) watchLeader(ctx context.Context, session *concurrency.Session
 			return
 		case <-ctx.Done():
 			return
+		case <-e.resignCh:
+			if e.evictLeader.Get() {
+				elec.Resign(ctx)
+				return
+			}
 		}
 	}
 }
 
+// EvictLeader set evictLeader to true, and this member can't be leader
+func (e *Election) EvictLeader() {
+	if e.evictLeader.Get() {
+		return
+	}
+
+	e.evictLeader.Set(true)
```

Comment:
How about using compare and swap instead?

Reply:
seems that `AtomicBool` dosen't support `CompareAndSwap` now  https://github.com/siddontang/go/blob/bdc77568d726a8702315ec4eafda030b6abc4f43/sync2/atomic.go#L134

Reply:
use another type like `int32` to simulate `bool`?

Reply:
change in b1fef8f